modules/cassandra/src/main/java/org/testcontainers/cassandra/CassandraContainer.java

```diff
@@ -0,0 +1,186 @@
+        Optional
+            .ofNullable(resourceLocation)
+            .map(MountableFile::forClasspathResource)
+            .ifPresent(mountableFile -> withCopyFileToContainer(mountableFile, pathNameInContainer));
+    }
+
+    /**
+     * Initialize Cassandra with the custom overridden Cassandra configuration
+     * <p>
+     * Be aware, that Docker effectively replaces all /etc/cassandra content with the content of config location, so if
+     * Cassandra.yaml in configLocation is absent or corrupted, then Cassandra just won't launch
+     *
+     * @param configLocation relative classpath with the directory that contains cassandra.yaml and other configuration files
+     */
+    public CassandraContainer withConfigurationOverride(String configLocation) {
+        this.configLocation = configLocation;
+        return self();
+    }
+
+    /**
+     * Initialize Cassandra with init CQL script
+     * <p>
+     *     CQL script will be applied after container is started (see using WaitStrategy).
+     * </p>
+     * <p>
+     *     If you override the Cassandra configuration (see {@link #withConfigurationOverride(String)}) to make the
+     *     authentication mandatory (using {@code PasswordAuthenticator} for the property {@code authenticator}), it's
+     *     strongly recommended to also use the {@link org.testcontainers.cassandra.wait.CassandraQueryWaitStrategy}
+     *     in order to guarantee the init script will be executed once the Cassandra node is really ready to execute
+     *     authenticated queries, otherwise you may encounter an error like this one:
+     *     "AuthenticationFailed('Failed to authenticate to 127.0.0.1: Error from server: code=0100 [Bad credentials]".
+     * </p>
+     *
+     * @param initScriptPath relative classpath resource
+     */
+    public CassandraContainer withInitScript(String initScriptPath) {
+        this.initScriptPath = initScriptPath;
+        return self();
+    }
+
+    /**
+     * Get username
+     *
+     * By default, Cassandra has authenticator: AllowAllAuthenticator in cassandra.yaml
+     * If username and password need to be used, then authenticator should be set as PasswordAuthenticator
+     * (through custom Cassandra configuration) and through CQL with default cassandra-cassandra credentials
+     * user management should be modified
+     */
+    public String getUsername() {
+        return USERNAME;
+    }
+
+    /**
+     * Get password
+     *
+     * By default, Cassandra has authenticator: AllowAllAuthenticator in cassandra.yaml
+     * If username and password need to be used, then authenticator should be set as PasswordAuthenticator
+     * (through custom Cassandra configuration) and through CQL with default cassandra-cassandra credentials
+     * user management should be modified
+     */
+    public String getPassword() {
+        return PASSWORD;
+    }
```

Comment:
what if username and password are different in cassandra.yaml?

Reply:
I kept these methods as is from the previous implementation.
But, we can't modify the default `cassandra` superuser, coming with these credentials, via the `cassandra.yaml` file. To modify these credentials, a user have to connect a first time with this superuser, then update the database superusers via a script (like the init script).
Source: https://cassandra.apache.org/doc/stable/cassandra/operating/security.html#password-authentication

So, should we keep these two methods? I don't really see any case where we could have different  values.

Reply:
ok, I think those are fine in order to build the cassandra's client